syntax = "proto3";
package SatelliteEarthSimulation;

option go_package = "github.com/Theossr/Helikos_test/api/proto";

message SimRequest {
    // TLE
    message TLE {
        string title = 1;
        string line1 = 2;
        string line2 = 3;
    }

    TLE tle = 1;
    uint64 start_epoch_ms = 2;
    float duration_s = 3;
    float step_s = 4;
    float h_fail_km = 5;
}

message SimResponse {

  // Frames
  message Frame {
    float t_rel_s = 1;

    message Geo {
      float lat_deg = 1;
      float lon_deg = 2;
      float alt_km = 3;
    }
    Geo geo = 2;

    message Mag {
      float x_nT = 1;
      float y_nT = 2;
      float z_nT = 3;
      float f_nT = 4;
    }
    Mag mag = 3;
  }
  repeated Frame frames = 1;

  // Decay
  message Decay {
    float threshold_km = 1;

    message TLEForward {
      bool crossed = 1;
      string time_utc = 2;
      double time_from_start_s = 3;
      float uncertainty_s = 4;
    }
    TLEForward tle_forward = 2;

    message PhysicsDrag {
      bool crossed = 1;
      string time_utc = 2;
      double time_from_start_s = 3;
      float uncertainty_s = 4;
    }
    PhysicsDrag physics_drag = 3;
  }
  Decay decay = 2;

  // Meta
  message Meta {
    uint32 points_returned = 1;
    uint32 points_generated = 2;
    uint32 skipped_frames = 3;
    string tle_epoch_utc = 4;
    bool out_of_tle_epoch_window = 5;
  }
  Meta meta = 3;
}

service SatelliteEarthSimulationService {
  rpc RunSimulation (SimRequest) returns (SimResponse);
}

